rust/src/dataset/scanner.rs

```diff
@@ -1826,12 +1826,13 @@ mod test {
                 schema.clone(),
                 vec![
                     Arc::new(Int32Array::from_iter_values(0..512)),
-                    Arc::new(vectors),
+                    Arc::new(vectors.clone()),
                 ],
             )
             .unwrap()];
 
-            let reader = RecordBatchIterator::new(batches.into_iter().map(Ok), schema.clone());
+            let reader =
+                RecordBatchIterator::new(batches.clone().into_iter().map(Ok), schema.clone());
```

Comment:
`batches.into_iter()` takes ownership without `clone()`, is it problematic here? I saw that you created another `batches` later.

Reply:
Good point.  The `clone()` is redundant because we don't use the same `batches` again.  I've removed it.